s3tests_boto3/functional/test_iam.py

```diff
@@ -0,0 +1,218 @@
+import json
+from operator import eq
+
+from botocore.exceptions import ClientError
+from nose.plugins.attrib import attr
+
+from s3tests.functional.utils import assert_raises
+from . import get_tenant_iam_client, get_tenant_user_id
+from .utils import _get_status
+
+
+@attr(resource='user-policy')
+@attr(method='put')
+@attr(operation='Verify Put User Policy')
+@attr(assertion='succeeds')
+@attr('user-policy')
+def test_put_user_policy_40447():
+    """TEST-40447"""
+    client = get_tenant_iam_client()
+
+    policy_document = json.dumps(
+        {"Version": "2012-10-17",
+         "Statement": {
+             "Effect": "Allow",
+             "Action": "*",
+             "Resource": "*"}}
+    )
+    client.put_user_policy(PolicyDocument=policy_document, PolicyName='AllAccessPolicy',
+                           UserName=get_tenant_user_id())
```

Comment:
PutUserPolicy is used to set a user policy by a user with user-policy caps on a different user. I believe the access key, secret access key that is being used to create iam_client is the same for the user returned by get_tenant_user_id()

Reply:
Also, what does this test check for? Success?. There is a utility method for put_user_policy in https://github.com/ceph/s3-tests/blob/master/s3tests_boto3/functional/test_sts.py#L80, that could be put in a common file and re-used for all put user policy tests.